### PR TITLE
Enable more govet checks and fix current issues in tests

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -64,14 +64,6 @@ linters-settings:
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Warnf
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Errorf
           - (github.com/golangci/golangci-lint/pkg/logutils.Log).Fatalf
-
-    # enable or disable analyzers by name
```
Comment (pjanotti) [inline .golangci.yml]: Do these conflicts with any of the other changes?

Comment (bogdandrutu) [reply .golangci.yml]: Not sure I understand.

Comment (pjanotti) [reply .golangci.yml]: why these are being removed? are the same covered by `govet`?

Comment (bogdandrutu) [reply .golangci.yml]: Actually the logic was to enable only `atomicalign` in `govet` if I understand it correctly. What I did was to keep all default enabled checks in `govet`

Comment (bogdandrutu) [reply .golangci.yml]: I read the source code, I was wrong Enable behavior is "enable + default". Fixed in https://github.com/open-telemetry/opentelemetry-collector/pull/739 by enabling all analyzers.

**Pull request closed without merging**